src/test/java/org/springframework/data/jpa/repository/query/ParameterBindingParserUnitTests.java

```diff
@@ -31,41 +32,42 @@ void identificationOfParameters() {
+		checkHasParameter(softly, ":id", true, "alone", false);
+		checkHasParameter(softly, "select something from x where id = :id", true, "named parameter", false);
+		checkHasParameter(softly, ":UPPERCASE", true, "uppercase", false);
+		checkHasParameter(softly, ":lowercase", true, "lowercase", false);
+		checkHasParameter(softly, ":2something", true, "beginning digit", false);
+		checkHasParameter(softly, ":2", true, "only digit", false);
+		checkHasParameter(softly, ":.something", true, "dot", false); // <--
+		checkHasParameter(softly, ":_something", true, "underscore", false);
+		checkHasParameter(softly, ":$something", true, "dollar", false); // <--
+		checkHasParameter(softly, ":\uFE0F", true, "non basic latin emoji", false); // <--
+		checkHasParameter(softly, ":\u4E01", true, "chinese japanese korean", false);
+		checkHasParameter(softly, "select something from x where id = ?1", true, "indexed parameter", false);
 
 		// <-- should we accept hash as named parameter start?
-		checkHasParameter(softly, "select something from x where id = #something", false, "hash");
+		checkHasParameter(softly, "select something from x where id = #something", false, "hash", false);
 
-		checkHasParameter(softly, "no bind variable", false, "no bind variable");
-		checkHasParameter(softly, ":\u2004whitespace", false, "non basic latin whitespace"); // <--
-		checkHasParameter(softly, "::", false, "double colon");
-		checkHasParameter(softly, ":", false, "end of query");
-		checkHasParameter(softly, ":\u0003", false, "non-printable");
-		checkHasParameter(softly, ":\u002A", false, "basic latin emoji");
-		checkHasParameter(softly, "\\:", false, "escaped colon");
-		checkHasParameter(softly, "::id", false, "double colon with identifier");
-		checkHasParameter(softly, "\\:id", false, "escaped colon with identifier");
+		checkHasParameter(softly, "no bind variable", false, "no bind variable", false);
+		checkHasParameter(softly, ":\u2004whitespace", false, "non basic latin whitespace", false); // <--
+		checkHasParameter(softly, "::", false, "double colon", false);
+		checkHasParameter(softly, ":", false, "end of query", false);
+		checkHasParameter(softly, ":\u0003", false, "non-printable", false);
+		checkHasParameter(softly, ":\u002A", false, "basic latin emoji", false);
+		checkHasParameter(softly, "\\:", false, "escaped colon", false);
+		checkHasParameter(softly, "::id", false, "double colon with identifier", false);
+		checkHasParameter(softly, "\\:id", false, "escaped colon with identifier", false);
 
 		softly.assertAll();
 	}
 
-	private void checkHasParameter(SoftAssertions softly, String query, boolean containsParameter, String label) {
+	private void checkHasParameter(SoftAssertions softly, String query, boolean containsParameter, String label,
+			boolean nativeQuery) {
 
-		StringQuery stringQuery = new StringQuery(query);
+		StringQuery stringQuery = new StringQuery(query, nativeQuery);
```

Comment:
This value is only ever set to `false`, so we can drop it.

Reply:
Oh ok! I thought it might be useful to have this option so in case new test cases come later, it can be easily extended. But I can remove it